api/src/Entity/Checklist.php

```diff
@@ -0,0 +1,165 @@
+class Checklist extends BaseEntity implements BelongsToCampInterface, CopyFromPrototypeInterface {
+    public const CAMP_SUBRESOURCE_URI_TEMPLATE = '/camps/{campId}/checklists.{_format}';
+
+    /**
+     * The camp this checklist belongs to.
+     */
+    #[ApiProperty(example: '/camps/1a2b3c4d')]
+    #[Groups(['read', 'create'])]
+    #[ORM\ManyToOne(targetEntity: Camp::class, inversedBy: 'checklists')]
+    #[ORM\JoinColumn(nullable: false, onDelete: 'cascade')]
+    public ?Camp $camp = null;
+
+    /**
+     * Copy contents from this source checklist.
+     */
+    #[ApiProperty(example: '/checklists/1a2b3c4d')]
+    #[Groups(['create'])]
+    public ?Checklist $copyChecklistSource;
+
+    /**
+     * All ChecklistItems that belong to this Checklist.
+     */
+    #[ApiProperty(writable: false, example: '["/checklist_items/1a2b3c4d"]')]
+    #[Groups(['read'])]
+    #[ORM\OneToMany(targetEntity: ChecklistItem::class, mappedBy: 'checklist', cascade: ['persist'])]
+    public Collection $checklistItems;
+
+    /**
+     * The human readable name of the checklist.
+     */
+    #[ApiProperty(example: 'PBS Ausbildungsziele')]
+    #[Groups(['read', 'write'])]
+    #[InputFilter\Trim]
+    #[InputFilter\CleanText]
+    #[Assert\NotBlank]
+    #[Assert\Length(max: 32)]
+    #[ORM\Column(type: 'text')]
+    public ?string $name = null;
+
+    public function __construct() {
+        parent::__construct();
+        $this->checklistItems = new ArrayCollection();
+    }
+
+    public function getCamp(): ?Camp {
+        return $this->camp;
+    }
+
+    /**
+     * @return ChecklistItem[]
+     */
+    public function getChecklistItems(): array {
+        return $this->checklistItems->getValues();
+    }
+
+    public function addChecklistItem(ChecklistItem $checklistItem): self {
+        if (!$this->checklistItems->contains($checklistItem)) {
+            $this->checklistItems[] = $checklistItem;
+            $checklistItem->checklist = $this;
+        }
+
+        return $this;
+    }
+
+    public function removeChecklistItem(ChecklistItem $checklistItem): self {
+        if ($this->checklistItems->removeElement($checklistItem)) {
+            // set the owning side to null (unless already changed)
+            if ($checklistItem->checklist === $this) {
+                $checklistItem->checklist = null;
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param Checklist $prototype
+     * @param EntityMap $entityMap
+     */
+    public function copyFromPrototype($prototype, $entityMap): void {
+        $entityMap->add($prototype, $this);
+
+        // copy Checklist base properties
+        $this->name = $prototype->name;
+
+        // deep copy ChecklistItems
+        foreach ($prototype->getChecklistItems() as $checklistItemPrototype) {
+            // deep copy root ChecklistItems
+            // skip non-root ChecklistItems as these are copyed by there parent
```

Comment:
is this necessary?
don't you have only root CheckListItems in this list?

Reply:
No. Each `CheckListItem` refs its `Checklist`.